res/controllers/DJ TechTools-MIDI Fighter Spectra-scripts.js

```diff
@@ -162,11 +162,18 @@ var MidiFighterSpectra;
                         }
 
                         // otherwise set the "dim" variant.
-                        components.HotcueButton.prototype.outputColor.bind(this)(colorCode);
+                        components.HotcueButton.prototype.outputColor.call(this, colorCode);
                     },
                 });
             }
         }
+
+        setCurrentDeck(newGroup) {
+            components.Deck.prototype.setCurrentDeck.call(this, newGroup);
+            for (const btn of MidiFighterSpectra.controller.selectDeck) {
+                btn.setActive(newGroup);
+            }
```

Comment:
just call `.trigger()` on the buttons. Potentially even the entire deck.

Reply:
We can't call trigger here because these are doing two fundamentally different things; trigger triggers the output for the connection, ie. pulse the LED (or stop it pulsing) at `end_of_track`. However, the `setActive` stuff is changing the brightness or color of the LED depending on if the deck is active or not, which isn't something we can connect to. We don't want one function handling both because we'd be sending a lot of unnecessary extra MIDI signals (which I suppose isn't the end of the world, but still isn't ideal). Ie if we swapped decks we'd send a "stop blinking" or "start blinking" signal even though that is already happening when all we wanted to do was dim the LED.